### ISSUE TITLE
[bug] cmake_layout generators folder path is not normalized

### ISSUE THREAD

**Issue by tiger-chan** [conan-io/conan#11795]
### Environment Details (include every applicable attribute)
  * Operating System+version: Windows 10/11
  * Compiler+version: N/A
  * Conan version: 1.51.0
  * Python version: 3.10.4

### Steps to reproduce (Include if Applicable)
1. Run `conan new bug/1.0 --template=cmake_exe`
2. Run `conan install .`

### Logs (Executed commands with output) (Include/Attach if Applicable)

```console
user@computer MINGW64 /d/dev/bug
$ conan new bug/1.0 --template=cmake_exe
File saved: CMakeLists.txt
File saved: conanfile.py
File saved: include/bug.h
File saved: src/bug.cpp
File saved: src/main.cpp
File saved: test_package/conanfile.py

user@computer MINGW64 /d/dev/bug
$ conan install . -pr:b=default
Configuration (profile_host):
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.runtime=MD
compiler.version=17
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

Configuration (profile_build):
[settings]
arch=x86_64
arch_build=x86_64
build_type=Release
compiler=Visual Studio
compiler.runtime=MD
compiler.version=17
os=Windows
os_build=Windows
[options]
[build_requires]
[env]

conanfile.py (bug/1.0): Installing package
Requirements
Packages

Installing (downloading, building) binaries...
conanfile.py (bug/1.0): Generator txt created conanbuildinfo.txt
conanfile.py (bug/1.0): Calling generate()
conanfile.py (bug/1.0): Aggregating env generators
conanfile.py (bug/1.0): Generated conaninfo.txt
conanfile.py (bug/1.0): Generated graphinfo
```

Resulting in a `CMakeUserPresets.json` with the following path.

```json
{
    "version": 4,
    "vendor": {
        "conan": {}
    },
    "include": [
        "D:\\dev\\bug\\build/generators\\CMakePresets.json"
    ]
}
```

The path should be normalized to the OS.  This appears to be caused by the following line.  As it is set but never normalized to the OS, it just assumes *nix file systems.
https://github.com/conan-io/conan/blob/release/1.51/conan/tools/cmake/layout.py#L34

**Comment by czoido** [comment]
Hi @tiger-chan,

Thanks a lot for reporting and the contribution. I'll add this for the next release.

**Comment by czoido** [comment]
Thansk for reporting!

Closed by https://github.com/conan-io/conan/pull/11809 to be released very soon in 1.51.1
